config/containerregistry/config.go

```diff
@@ -17,6 +17,16 @@ func Configure(p *config.Provider) {
 		r.LateInitializer = config.LateInitializer{
 			IgnoredFields: []string{"encryption"},
 		}
+
+		r.TerraformResource.Schema["retention_policy"].MaxItems = 1
+		r.TerraformResource.Schema["trust_policy"].MaxItems = 1
+		r.TerraformResource.Schema["encryption"].MaxItems = 1
+		r.TerraformResource.Schema["network_rule_set"].MaxItems = 1
+
+		r.AddSingletonListConversion("retention_policy", "retention_policy")
```

Comment:
Looks like the CRD paths are misconfigured, i.e.:
```suggestion
		r.AddSingletonListConversion("retention_policy", "retentionPolicy")
```
We also need to fix the other snake_case second arguments to `Resource.AddSingletonListConversion`.

Reply:
Done!